### PR TITLE
Add trivy image scanning

### PR DESCRIPTION
**Pull request by sjpb**
- Scans builds for vulerabilities using Trivy. Fails build if CRITICAL vulns found
- Refactors builds to only build one image per matrix run, and only run the builds we use.

### CODE REVIEW

**Comment by bertiethorpe** [comment]
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/10321752874

**Comment by bertiethorpe** [comment]
https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/10370481508

File: dev/setup-env.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 set -euo pipefail
 
 if [[ ! -d "venv" ]]; then
-    /usr/bin/python3.8 -m venv venv # use `sudo yum install python38` on Rocky Linux 8 to install this
+    /usr/bin/python3 -m venv venv # use `sudo yum install python38` on Rocky Linux 8 to install this
```
Comment (sjpb) [inline dev/setup-env.sh:6]: Is this OK on both RL8 and RL9? I.e. we end up with a working python version which supports selinux?

File: environments/.stackhpc/hooks/post.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- hosts: openondemand
+  become: yes
+  gather_facts: false
+  tasks:
+    - name: Delete trivy vulnerable files
```
Comment (sjpb) [inline environments/.stackhpc/hooks/post.yml:5]:
```suggestion
    - name: Delete ondemand files causing Trivy scan false-positives
      # Raised at https://github.com/OSC/ondemand/security/advisories/GHSA-f7j8-ppqm-m5vw
      # All declared not to be an issue by Open Ondemand as relevant packages not installed
```

**Review by sjpb — COMMENTED** [review]
LGTM other than a task name / comments change please.

**Review by bertiethorpe — APPROVED** [review]

**Review by bertiethorpe — APPROVED** [review]

**Pull request merged**